Makefile

```diff
@@ -0,0 +1,47 @@
+BINDIR=/usr/local/bin
+UNITDIR=/etc/systemd/system
+SYSCONFDIR=/usr/local/etc
+DEFAULTDIR=$(SYSCONFDIR)/default
+DESTDIR=
+PROGNAME=alertmanager_matrix
+
+all: bin/$(PROGNAME) $(PROGNAME).service
+
+ROOT_DIR := $(shell dirname $(realpath $(firstword $(MAKEFILE_LIST))))
+
+.PHONY: clean dist rpm srpm install-unit install-prog install-default install
+
+clean:
+	cd $(ROOT_DIR) || exit $$? ; find -name '*.pyc' -o -name '*~' -print0 | xargs -0 rm -f
+	cd $(ROOT_DIR) || exit $$? ; rm -rf *.tar.gz *.rpm
+	cd $(ROOT_DIR) || exit $$? ; rm -rf bin $(PROGNAME).service
+
+dist: clean
+	@which rpmspec || { echo 'rpmspec is not available.  Please install the rpm-build package with the command `dnf install rpm-build` to continue, then rerun this step.' ; exit 1 ; }
+	cd $(ROOT_DIR) || exit $$? ; excludefrom= ; test -f .gitignore && excludefrom=--exclude-from=.gitignore ; DIR=`rpmspec -q --queryformat '%{name}-%{version}\n' *spec | head -1` && FILENAME="$$DIR.tar.gz" && tar cvzf "$$FILENAME" --exclude="$$FILENAME" --exclude=.git --exclude=.gitignore $$excludefrom --transform="s|^|$$DIR/|" --show-transformed *
```

Comment:
I would expect that the `dist` target be platform-agnostic, is it possible to remove the `rpmspec` requirement in this target?

Reply:
Yes.  I have pushed a commit that does exactly this, by using a new shell script which falls back to `git describe` to detect which version to publish.